src/Decompiler/Analysis/CallApplicationBuilder.cs

```diff
@@ -59,18 +60,30 @@ public CallApplicationBuilder(SsaState ssaCaller, Statement stmCall, CallInstruc
 
         public override Expression Bind(Identifier id)
         {
-            return With(uses, id.Storage);
+            return WithUses(id.Storage);
         }
 
         public override OutArgument BindOutArg(Identifier id)
         {
-            var exp = With(defs, id.Storage);
+            var exp = WithDefinitions(id.Storage);
             return new OutArgument(arch.FramePointerType, exp);
         }
 
         public override Expression BindReturnValue(Identifier id)
         {
-            return With(defs, id.Storage);
+            return WithDefinitions(id.Storage);
+        }
+
+        private Expression WithUses(Storage stg)
+        {
+            this.bindUses = true;
+            return With(uses, stg);
+        }
+
+        private Expression WithDefinitions(Storage stg)
```

Comment:
Could you add a comment why we don't want to bind uses in this case?

Reply:
>why we don't want to bind uses in this case?

Because this this is return, so we should bind it with `defs`, is not it? Do you want me to add such comment before `WithDefinitions` method?